### ISSUE TITLE
Cannot resolve a Slate point from DOM point: [object Text],10

### ISSUE THREAD

**Issue by robinjoseph08** [ianstormtaylor/slate#4871]
**Description**
With a custom inline void element, clicking on the element sometimes produces the following error:

```
index.es.js:2876 Uncaught Error: Cannot resolve a Slate point from DOM point: [object Text],10
    at Object.toSlatePoint (index.es.js:2876:1)
    at Object.toSlateRange (index.es.js:2933:1)
    at HTMLDocument.<anonymous> (index.es.js:1188:1)
    at invokeFunc (debounce.js:95:1)
    at leadingEdge (debounce.js:105:1)
    at HTMLDocument.debounced (debounce.js:172:1)
    at invokeFunc (debounce.js:95:1)
    at trailingEdge (debounce.js:144:1)
    at timerExpired (debounce.js:132:1)
```

**Recording**

Here's a recording of the issue happening in my codebase:

https://user-images.githubusercontent.com/2454505/157132522-69d1fd3e-29ae-475a-8ebd-5b57a1979d6f.mov

**Sandbox**
Link to a stripped-down sandbox to reproduce the issue: https://codesandbox.io/s/cannot-resolve-a-slate-point-error-tzg8bi?file=/src/App.tsx

**Steps**
To reproduce the behavior:
1. In the sandbox, click on the blue `Token` element within the editor. That should cause the error to be thrown.

**Expectation**
In my codebase, the error doesn't seem to be affecting anything (from what I can tell), but it's throwing errors very frequently. I expect the error to not get thrown.

**Environment**
- Slate Version: 0.73.1
- Slate React Version: 0.74.0
- Operating System: macOS 12.2.1
- Browser: Chrome 98.0.4758.109

**Context**
I found this issue after upgrading from `slate-react` 0.65.2 to 0.74.0. I _think_ it might be related to [this change](https://github.com/ianstormtaylor/slate/pull/4132), but I don't know enough about Slate internals to be sure.

**Comment by bendman** [comment]
I've had similar issues, and it seems like this error occurs when you render a void element without adding `contentEditable={false}` to it.

I've been manually adding this property to my void elements, but it seems like if this is required and if so this prop should probably get added automatically to the `props.attributes` of void element render functions.

**Comment by robinjoseph08** [comment]
@bendman thanks for the tip! It looks like that fixed it in the sandbox, and helped a little in my actual codebase (now when I click on the void element when it's readonly, it doesn't throw the error), but it's still there when it's not readonly. That definitely means that the sandbox wasn't accurately reflecting my problem. But this is good progress! I'll poke at the sandbox a bit more and see if I can more accurately reflect my issue.

**Comment by robinjoseph08** [comment]
Ah okay, I had missed adding the Token plugin to the editor, so things were in a weird state. By wrapping it in a `withTokens` call, this is feeling much more consistent with the issue that I have in my actual codebase. Sorry about the initial confusion!

The sandbox has been updated, and it's at the same link: https://codesandbox.io/s/cannot-resolve-a-slate-point-error-tzg8bi?file=/src/App.tsx

And even with `contentEditable={false}`, it's still throwing the error.

**Comment by robinjoseph08** [comment]
Okay, in poking around it more, I was able to fix the issue.

It looks like it stemmed from the fact that I was manually rendering my `Leaf` component with props of type `RenderLeafProps`. That type requires an `attributes` prop of value `{ "data-slate-leaf": true }`. I'm reusing the `Leaf` component since, in my codebase, it's got rich text styling on it, so I didn't want to have to reimplement that for Tokens too. So because I was manually setting that `attribute`, it was assuming something about the structure of the DOM which wasn't being satisfied because I was just manually adding it without considering those assumptions.

The only reason I added it was because of the types, so I just modified my `Leaf` component to not require `attributes`, and then I removed the manual adding of `attributes`, and the error stopped occurring!

Thanks for the help, and maybe this will help anyone that is also reusing their `Leaf` component.
